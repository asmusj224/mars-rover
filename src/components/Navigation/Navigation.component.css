.navigation {
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin: 0 2rem; }
  .navigation__list {
    display: flex;
    list-style: none; }
  .navigation__item {
    font-size: 13px;
    margin-right: 1rem; }
  .navigation__link {
    text-decoration: none;
    opacity: 0.8; }
    .navigation__link:hover {
      opacity: 1; }
    .navigation__link--gray {
      color: #a9a9a9; }
    .navigation__link--black {
      color: #2d2d2d; }
