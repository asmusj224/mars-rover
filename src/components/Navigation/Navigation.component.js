import React from 'react';
import { Link } from 'react-router-dom';

import './Navigation.component.css';
import * as routes from '../../constants/routes';

const Navigation = () => (
    <div className="navigation">
        <h2><Link className="navigation__link navigation__link--black" to={routes.IMAGE_OF_THE_DAY}>Mars Rover</Link></h2>
        <ul className="navigation__list">
            <li className="navigation__item"><Link className="navigation__link navigation__link--gray" to={routes.IMAGE_OF_THE_DAY}>Image of the day</Link></li>
            <li className="navigation__item"><Link className="navigation__link navigation__link--gray" to={routes.IMAGE_GALLERY}>Image Gallery</Link></li>
        </ul>
    </div>
);

export default Navigation;;