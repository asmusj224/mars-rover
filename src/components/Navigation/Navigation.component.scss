@import 'styles/_colors.scss';
.navigation {
    display: flex;
    align-items: center;
    justify-content: space-between;
    margin: 0 2rem;
    &__list {
        display: flex;
        list-style: none;
    }
    &__item {
        font-size: 13px;
        margin-right: 1rem;
    }
    &__link {
        text-decoration: none;
        opacity: 0.8;
        &:hover {
            opacity: 1;
        }
        &--gray {
            color: $darkgray;
        }
        &--black {
            color: $black;
        }
    }
}