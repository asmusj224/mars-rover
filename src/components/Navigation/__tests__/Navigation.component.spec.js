import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router'

import Navigation from '../Navigation.component';

const context = { router: { isActive: (a, b) => true } };


describe('<Navigation />', () => {
    it('renders correctly', () => {
        const component = renderer.create(
            <StaticRouter context={context}>
                <Navigation />
            </StaticRouter>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

