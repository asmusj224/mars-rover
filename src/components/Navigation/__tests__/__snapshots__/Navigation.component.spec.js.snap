// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> renders correctly 1`] = `
<div
  className="navigation"
>
  <h2>
    <a
      className="navigation__link navigation__link--black"
      href="/image-of-the-day"
      onClick={[Function]}
    >
      Mars Rover
    </a>
  </h2>
  <ul
    className="navigation__list"
  >
    <li
      className="navigation__item"
    >
      <a
        className="navigation__link navigation__link--gray"
        href="/image-of-the-day"
        onClick={[Function]}
      >
        Image of the day
      </a>
    </li>
    <li
      className="navigation__item"
    >
      <a
        className="navigation__link navigation__link--gray"
        href="/image-gallery"
        onClick={[Function]}
      >
        Image Gallery
      </a>
    </li>
  </ul>
</div>
`;
